dashboard/lib/services/curriculum_pdfs/resources.rb

```diff
@@ -191,14 +204,56 @@ def fetch_url_to_path(url, path)
           return nil
         end
 
-        # Returns a GoogleDrive::File object for the given url
-        #
-        # @see https://www.rubydoc.info/gems/google_drive/GoogleDrive/File
-        def google_drive_file_by_url(url)
-          @google_drive_session ||= GoogleDrive::Session.from_service_account_key(
-            StringIO.new(CDO.gdrive_export_secret&.to_json || "")
-          )
-          return @google_drive_session.file_by_url(url)
+        # url_to_id(url) is borrowed from GoogleDrive api
+        # The license of this software is "New BSD Licence".
+        # Copyright (c) 2013, Hiroshi Ichikawa <http://gimite.net/>
+        # Copyright (c) 2013, David R. Albrecht <https://github.com/eldavido>
+        # Copyright (c) 2013, Guy Boertje <https://github.com/guyboertje>
+        # Copyright (c) 2013, Phuogn Nguyen <https://github.com/phuongnd08>
+        # All rights reserved.
+
+        # Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
```

Comment:
nit: I _think_ you can delete this bit down (but keep the link because it's helpful). But not worth a rebuild imo